### DISCUSSION TITLE
Multiple heat sources

### DISCUSSION THREAD

**Discussion by FatimahDzr** [idaholab/moose#24806] [Q&A Modules: General]
I want to add two heat sources at a same time to my geometry. at top I have laser and from both right and left sides I wanna apply the electric field. So currently I'm able to separately apply the laser heat source and also the joule heating effect by electric field. but I don't know how can I combine both of them as a combined heat source in my problem. here is my kernels: 
```
# 1: thermal conduction
# rou*cp*dT/dt=grad(K)*grad(T)+qdot
[Kernels]
  #grad(K)*grad(T)
  [heat_conduction]
    type = ADHeatConduction
    variable = T
  []
  #rou*cp*dT/dt
  [time_derivative]
    type = ADHeatConductionTimeDerivative
    variable = T
  []
  # 2: laser
  #qdot
  [heat_source1]
    type = ADMatHeatSource
    material_property = volumetric_heat
    variable = T
  []
  # 3: electrical conduction
  #qdot
[heat_source2]
  type = ADJouleHeatingSource
  elec = phi
  variable = T
[]
# grad(sigma*grad(phi))
  [./electric]
    type = ADHeatConduction
    variable = phi
    thermal_conductivity = electrical_conductivity
  []
[]


and here is my error:
 Solve Did NOT Converge!
  Finished Solving                                                                       [ 24.17 s] [    0 MB]
Aborting as solve did not converge


*** ERROR ***
The following error occurred in the object "TimeStepper", of type "ConstantDT".

Solve failed and timestep already at or below dtmin, cannot continue!

```

I would be grateful if someone help me with this issue that how can I combine two heat sources together.

**Comment by GiudGiud** [comment]
Hello

it s ok to have two kernels, once for each heat source.

The problem is likely that either:
- the problem is ill posed for another reason
- the numerical solver is failing to find the solution

Please see this page for more help
https://mooseframework.inl.gov/application_usage/failed_solves.html
Guillaume
